app/Jobs/CreateCustomerJob.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace App\Jobs;
+
+use App\Models\Customer;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+
+class CreateCustomerJob implements ShouldQueue
+{
+    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
+
+    /**
+     * Create a new job instance.
+     */
+    protected $data;
+
+    public function __construct(array $data)
+    {
+        //
+        $this->data = $data;
+    }
+
+    /**
+     * Execute the job.
+     */
+    public function handle()
+    {
+        // Create a new customer using the provided data
```

Comment:
the aim of jobs is to push created data to other services, please look at other publishing job and how they are used, and when dispatching we point it to a particular , we usually do that via a look based on the env and nnpcreusable service config